code/scripting/api/objs/ship.cpp

```diff
@@ -1008,6 +1008,30 @@ ADE_FUNC(turnTowardsPoint,
 	return ADE_RETURN_NIL;
 }
 
+ADE_FUNC(turnTowardsOrientation,
+	l_Ship,
+	"orientation target, [boolean respectDifficulty = true, vector turnrateModifier /* 100% of tabled values in all rotation axes by default */]",
+	"turns the ship towards the specified orientation during this frame",
+	nullptr,
+	nullptr)
+{
+	object_h ship;
+	matrix_h* target;
+	bool diffTurn = true;
+	vec3d* modifier = nullptr;
+
+	int argnum = ade_get_args(L, "oo|bo", l_Ship.Get(&ship), l_Matrix.GetPtr(&target), &diffTurn, l_Vector.GetPtr(&modifier));
+	if (argnum == 0) {
+		return ADE_RETURN_NIL;
+	}
+
+	matrix* mat = target->GetMatrix();
+	vec3d targetVec = mat->vec.fvec + ship.objp->pos;
```

Comment:
I think this would be the part that you could encapsulate.  Something like `vm_find_goal_fvec(vec3d* dest, matrix* mat, vec3d* pos)
{
*dest = mat->vec.fvec + pos;
}`

The point in making this function is to standardize handling, reduce copy and paste errors, and make the method more easily discoverable. But it's a small difference, so the utility in creating it is not as high, imo.  I'm hitting approve.

Reply:
Good point, that makes sense. @Goober5000 does encapsulating in that way sound good to you as well?

Reply:
Hmm. I was thinking that both functions could delegate to a single helper function, but that `l_Ship` vs `l_AI_Helper` throws a wrench into things.  Plus, `turnTowardsPoint` is already duplicated.

That suggested function would be nice, but it's such a small change it may not be worth it.  As said, the utility is not high.

I'll approve as-is, no changes required.

Reply:
Sounds good. Yeah, if in the future a PR to encapsulate both is warranted by the addition of more LUA AI and plain ship AI modes then that might be useful, but for now I'll follow the previous lead and keep as is. Thanks for the discussion!